From 2fb451115aa431f63030c1d105277e6f57cc0659 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Thu, 17 Mar 2022 15:19:31 +0800
Subject: [PATCH 084/274] LF-6254-10 arm64: dts: imx93-11x11-evk: Add RM67199
 MIPI DSI panel support

This patch adds Raydium RM67199 MIPI DSI panel support
on i.MX93 11x11 EVK platform.

Cc: Sandor Yu <Sandor.yu@nxp.com>
Reviewed-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Liu Ying <victor.liu@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |  2 +-
 .../dts/freescale/imx93-11x11-evk-rm67199.dts | 52 +++++++++++++++++++
 2 files changed, 53 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/imx93-11x11-evk-rm67199.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index d8a134fd8..4cf060651 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -244,6 +244,6 @@ dtb-$(CONFIG_ARCH_MXC) += imx8qm-mek-sof-cs42888.dtb imx8qm-mek-sof-wm8960.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8dxl-evk-root.dtb imx8dxl-evk-inmate.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx93-11x11-evk.dtb imx93-11x11-evk-rpmsg.dtb \
 			  imx93-11x11-evk-lpspi.dtb imx93-11x11-evk-lpspi-slave.dtb \
-			  imx93-11x11-evk-i3c.dtb
+			  imx93-11x11-evk-i3c.dtb imx93-11x11-evk-rm67199.dtb
 dtb-$(CONFIG_ARCH_S32) += s32v234-evb.dtb \
 			  s32v234-sbc.dtb
diff --git a/arch/arm64/boot/dts/freescale/imx93-11x11-evk-rm67199.dts b/arch/arm64/boot/dts/freescale/imx93-11x11-evk-rm67199.dts
new file mode 100644
index 000000000..f66e205e6
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx93-11x11-evk-rm67199.dts
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2022 NXP
+ */
+
+#include "imx93-11x11-evk.dts"
+
+&adv7535 {
+	status = "disabled";
+};
+
+/delete-node/ &dsi_to_adv7535;
+/delete-node/ &adv7535_to_dsi;
+
+&dsi {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	panel@0 {
+		compatible = "raydium,rm67199";
+		reg = <0>;
+		reset-gpio = <&adp5585gpio 6 GPIO_ACTIVE_LOW>;
+		dsi-lanes = <4>;
+		video-mode = <2>;	/* 0: burst mode
+					 * 1: non-burst mode with sync event
+					 * 2: non-burst mode with sync pulse
+					 */
+		width-mm = <68>;
+		height-mm = <121>;
+		status = "okay";
+
+		port {
+			panel_in: endpoint {
+				remote-endpoint = <&dsi_out>;
+			};
+		};
+	};
+
+	ports {
+		port@1 {
+			reg = <1>;
+
+			dsi_out: endpoint {
+				remote-endpoint = <&panel_in>;
+			};
+		};
+	};
+};
+
+&lcdif {
+	assigned-clock-rates = <484000000>, <121000000>, <400000000>, <133333333>;
+};
-- 
2.25.1

